dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageLoader.java

```diff
@@ -197,6 +186,15 @@ public InputStream buildStream(IHTMLPage htmlPage, PageMode mode, final String f
 
     }
 
+    private void addWidgetPreExecuteCodeAndPageInfo(final HTMLPageAsset htmlPage, final PageMode mode,
+            final StringBuilder stringBuilder, final User user) throws DotSecurityException, DotDataException {
+        final PageRenderUtil pce = new PageRenderUtil(htmlPage, user, mode);
+        // Add the pre-execute code of a widget to the page
+        stringBuilder.append(pce.getWidgetPreExecute());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5178716)